### PR TITLE
Update To `aws-sdk` v3 And Node v20

### PR DESCRIPTION
**Pull request by JamieB-gu**
Reinstates #38, as discussed in #46, with an upgrade to `aws-sdk` v3.

Similar to https://github.com/guardian/email-rendering/pull/387

It would be nice to get #45 done first so this can be deployed easily.

## Changes

- Started with the codemod suggested by the docs and @abeddow91 : https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/migrating-to-v3.html

- Removed global region code; the codemod mentions that's not supported any more.

- Stopped wrapping credentials in a `Credentials` constructor, as the types suggest clients use a plain object, as does this docs example: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-api.html

- Updated @zeit/ncc to latest @vercel/ncc. The bundle script no longer works due to newer JS syntax and changes in node. Upgrading means switching to the vercel version because the package has been deprecated/moved:

  - https://www.npmjs.com/package/@zeit/ncc
  - https://github.com/zeit/ncc

- Updated the `promisify` test function. The `aws-sdk` v3 returns promises from API calls by default, without needing the `promise` method. Therefore the mocks should also do this.

- Changed the handling of an S3 `GetObject` body, as this used to be a Node.js `Buffer` but is now an AWS `SdkStream`:
  - v2: https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#getObject-property
  - v3: https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-client-s3/Interface/GetObjectCommandOutput/

## Testing

Deployed to CODE using this branch, which includes the changes from #45 to make it easier to deploy: https://github.com/guardian/front-pressed-lambda/tree/aws-sdk-v3-CODE-testing

Monitoring on the lambda shows a stable success rate at 100%, and log messages suggest it's processing records successfully.

### CODE REVIEW

File: lambda/index.js [hunk 1]
```diff
@@ -13,34 +16,39 @@ const MAX_INCIDENT_LENGTH = 250;
...
+    sessionToken: stsCredentials.SessionToken
+  };
+  const dynamo = new DynamoDB({
+    credentials: assumedCredentials,
+    region: config.AWS.region
+  });
+  const lambda = new Lambda({
+    credentials: assumedCredentials,
+    region: config.AWS.region
+  });
```
Comment (JamieB-gu) [inline lambda/index.js:51]: Would appreciate any feedback on this change.

I think the result is much the same as using this API described in the docs? Perhaps switching to that would be preferable? https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-credential-providers/#fromtemporarycredentials

File: package.json [hunk 2]
```diff
@@ -9,10 +9,13 @@
   "isAwsLambda": true,
   "riffraffFile": "./riff-raff.yaml",
   "devDependencies": {
+    "@aws-sdk/client-dynamodb": "^3.489.0",
+    "@aws-sdk/client-lambda": "^3.489.0",
+    "@aws-sdk/client-s3": "^3.489.0",
+    "@aws-sdk/client-sts": "^3.489.0",
```
Comment (mxdvl) [inline package.json:15]: Our recommendation is to use pinned versions
```suggestion
    "@aws-sdk/client-dynamodb": "3.489.0",
    "@aws-sdk/client-lambda": "3.489.0",
    "@aws-sdk/client-s3": "3.489.0",
    "@aws-sdk/client-sts": "3.489.0",
```

File: lambda/index.js (hunk 1 above)
Comment (mxdvl) [reply lambda/index.js:51]: Why not pass the `stsCredentials` directly?

Comment (JamieB-gu) [reply lambda/index.js:51]: They're different types: the first is of type `Credentials` and the second of type `AwsCredentialIdentity`. They do have different numbers of keys, but the keys also have different names (TitleCase vs camelCase).

I'm also trying to minimise the number of code changes, as ultimately this is a step to getting #50 merged.

File: package.json (hunk 2 above)
Comment (JamieB-gu) [reply package.json:15]: It is, but that's not something that this repo has adopted by the looks, so the other dependencies aren't pinned like this. We could update the whole file to pinned deps in a follow-up so we don't end up with a mix?

**Review by mxdvl — APPROVED** [review]

**Comment by JamieB-gu** [comment]
Seeing some errors in CODE, so moving into draft.

**Comment by JamieB-gu** [comment]
Moving back to review after [9a53784](https://github.com/guardian/front-pressed-lambda/pull/49/commits/9a537846fbc5d54dd7b54841f4eb284997980be2), monitoring on CODE now shows stable success rate.

**Review by mxdvl — APPROVED** [review]

**Pull request merged**